glide-core/redis-rs/redis/src/cluster_async/mod.rs

```diff
@@ -2139,6 +2204,319 @@ where
             .map_err(|err| (OperationTarget::Node { address }, err))
     }
 
+    // This function handles commands with routing info of MultiSlot (like MSET or MGET), creates sub-commands for the matching slots and add it to the correct pipeline
+    async fn handle_multi_slot_routing(
+        pipelines_by_connection: &mut NodePipelineMap<C>,
+        core: Core<C>,
+        cmd: &Cmd,
+        index: usize,
+        slots: Vec<(Route, Vec<usize>)>,
+    ) {
+        // inner_index is used to keep track of the index of the sub-command inside cmd
+        for (inner_index, (route, indices)) in slots.iter().enumerate() {
+            let conn = {
+                let lock = core.conn_lock.read().expect(MUTEX_READ_ERR);
+                lock.connection_for_route(route)
+            };
+            if let Some((address, conn)) = conn {
+                let new_cmd =
+                    crate::cluster_routing::command_for_multi_slot_indices(cmd, indices.iter());
```

Comment:
add `use crate::cluster_routing::command_for_multi_slot_indices` and remove the prefix